### PR TITLE
Add tool to bulk-create issues.

### PR DESCRIPTION
**Pull request by potiuk**
Simple tool to create issues in bulk from CSV file.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: dev/prepare_bulk_issues.py [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+
+def render_template_file(
+    template_name: str,
+    context: Dict[str, Any],
+    autoescape: bool = True,
+    keep_trailing_newline: bool = False,
+) -> str:
+    """
+    Renders template based on it's name. Reads the template from <name> filein current dir.
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
    Renders template based on its name. Reads the template from <name> file in the current dir.
```

File: dev/prepare_bulk_issues.py [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+
+
+def render_template_string(
+    template_string: str,
+    context: Dict[str, Any],
+    autoescape: bool = True,
+    keep_trailing_newline: bool = False,
+) -> str:
+    """
+    Renders template based on it's name. Reads the template from <name> filein current dir.
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
    Renders template based on its name. Reads the template from <name> file in the current dir.
```

File: dev/prepare_bulk_issues.py [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+
+option_github_token = click.option(
+    "--github-token",
+    type=str,
+    required=True,
+    help=textwrap.dedent(
+        """
+        Github token used to authenticate.
+        You can set omit it if you have GITHUB_TOKEN env variable set
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
        You can omit it if you have GITHUB_TOKEN env variable set
```

File: dev/prepare_bulk_issues.py [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+        Can be generated with:
+        https://github.com/settings/tokens/new?description=Write%20sssues&scopes=repo:status,public_repo"""
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
        https://github.com/settings/tokens/new?description=Write%20issues&scopes=repo:status,public_repo"""
```

File: dev/prepare_bulk_issues.py [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+option_verbose = click.option(
+    "--verbose",
+    is_flag=True,
+    help="Print verbose information about performed steps",
+)
+
+option_dry_run = click.option(
+    "--dry-run",
+    is_flag=True,
+    help="Do not create issues, just print the issues to create them",
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
    help="Do not create issues, just print the issues to be created",
```

File: dev/prepare_bulk_issues.py [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+    type=str,
+    default="apache/airflow",
+    help="Repo to use",
+)
+
+option_title = click.option(
+    "--title",
+    type=str,
+    required="true",
+    help="Title of the issue to create (might contain {{ name }}) to indicate the name of the group",
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
    help="Title of the issues to create (might contain {{ name }} to indicate the name of the group)",
```

File: dev/prepare_bulk_issues.py [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+)
+
+option_labels = click.option(
+    "--labels",
+    type=str,
+    help="Labels to assign to the issues (coma-separated)",
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]: Funny typo 😆 
```suggestion
    help="Labels to assign to the issues (comma-separated)",
```

File: dev/prepare_bulk_issues.py [hunk 8]
```diff
@@ -0,0 +1,255 @@
...
+                    processed_issues += 1
+                    sleep(2)  # avoid secondary rate limit!
+                    if max_issues is not None:
+                        max_issues -= 1
+                        if max_issues == 0:
+                            break
+            except GithubException as e:
+                console.print(f"[red]Error!: {e}[/]")
+                console.print(
+                    f"[yellow]Restart with `--start-from {processed_issues+start_from}` " f"to continue.[/]"
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
                    f"[yellow]Restart with `--start-from {processed_issues+start_from}` to continue.[/]"
```

File: dev/prepare_bulk_issues.py [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+                max_issues -= 1
+                if max_issues == 0:
+                    break
+        console.print()
+        console.print(f"Displayed {processed_issues} issue(s).")
+    else:
+        labels_list: List[str] = labels.split(",") if labels else []
+        issues_to_create = int(min(total_issues, max_issues if max_issues is not None else total_issues))
+        with Progress(console=console) as progress:
+            task = progress.add_task(f"Creating {issues_to_create} issues", total=issues_to_create)
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
            task = progress.add_task(f"Creating {issues_to_create} issue(s)", total=issues_to_create)
```

**Review by mnojek — COMMENTED** [review]
Very useful tool!
No big concerns, just a few typos.

**Review by mnojek — APPROVED** [review]

File: dev/prepare_bulk_issues.py [hunk 10]
```diff
@@ -0,0 +1,262 @@
...
+
+console = Console(width=400, color_system="standard")
+
+MY_DIR_PATH = os.path.dirname(__file__)
+SOURCE_DIR_PATH = os.path.abspath(os.path.join(MY_DIR_PATH, os.pardir))
+
+
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1
+#    NOTE! GitHub has secondary rate limits for issue creation, and you might be
+#          temporary blocked from creating issues and PRs if you create too many
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
#          temporarily blocked from creating issues and PRs if you create too many
```

File: dev/prepare_bulk_issues.py [hunk 11]
```diff
@@ -0,0 +1,262 @@
...
+
+logger = logging.getLogger(__name__)
+
+console = Console(width=400, color_system="standard")
+
+MY_DIR_PATH = os.path.dirname(__file__)
+SOURCE_DIR_PATH = os.path.abspath(os.path.join(MY_DIR_PATH, os.pardir))
+
+
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]: Is it an easter egg? 🐰 
```suggestion
# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

File: dev/prepare_bulk_issues.py [hunk 12]
```diff
@@ -0,0 +1,262 @@
...
+#    NOTE! GitHub has secondary rate limits for issue creation, and you might be
+#          temporary blocked from creating issues and PRs if you create too many
+#          issues in a short time
+# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1
```
Comment (mnojek) [inline dev/prepare_bulk_issues.py]:
```suggestion
# !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

**Review by mnojek — APPROVED** [review]

**Review by eladkal — APPROVED** [review]

**Pull request merged**